packages/core/core/src/Transformation.js

```diff
@@ -492,36 +511,52 @@ export default class Transformation {
     filePath: FilePath,
     isSource: boolean,
     pipeline: ?string,
+    inputEnv: InternalEnvironment,
   ): Promise<Pipeline> {
+    let transformers = await this.parcelConfig.getTransformers(
+      filePath,
+      pipeline,
+      this.request.isURL,
+    );
+
+    // Allow transformers in the pipeline to override the requested environment.
+    // For example, some transformers may not be affected by certain options
+    // (e.g. library mode, scope hoisting, or output formats).
+    let env = transformers.reduce((env, {plugin}) => {
+      if (typeof plugin.getEnvironment === 'function') {
```

Comment:
Nit: so far, something like `if(plugin.getEnvironment)` has been used in this file for checking optional methods